timemachine/cpp/src/kernels/k_nonbonded.cuh

```diff
@@ -828,34 +882,31 @@ void __global__ k_nonbonded_exclusions(
 
         }
 
-        gi_x += FLOAT_TO_FIXED(-delta_prefactor*delta_x);
-        gi_y += FLOAT_TO_FIXED(-delta_prefactor*delta_y);
-        gi_z += FLOAT_TO_FIXED(-delta_prefactor*delta_z);
+        gi_x -= FLOAT_TO_FIXED_NONBONDED(delta_prefactor*delta_x);
+        gi_y -= FLOAT_TO_FIXED_NONBONDED(delta_prefactor*delta_y);
+        gi_z -= FLOAT_TO_FIXED_NONBONDED(delta_prefactor*delta_z);
 
-        gj_x += FLOAT_TO_FIXED(delta_prefactor*delta_x);
-        gj_y += FLOAT_TO_FIXED(delta_prefactor*delta_y);
-        gj_z += FLOAT_TO_FIXED(delta_prefactor*delta_z);
+        gj_x -= FLOAT_TO_FIXED_NONBONDED(-delta_prefactor*delta_x);
+        gj_y -= FLOAT_TO_FIXED_NONBONDED(-delta_prefactor*delta_y);
+        gj_z -= FLOAT_TO_FIXED_NONBONDED(-delta_prefactor*delta_z);
 
-        du_dl_i += FLOAT_TO_FIXED(-delta_prefactor*delta_w*lambda_offset_i*cutoff);
-        du_dl_j += FLOAT_TO_FIXED(delta_prefactor*delta_w*lambda_offset_j*cutoff);
+        du_dl_i -= FLOAT_TO_FIXED_NONBONDED(delta_prefactor*delta_w*lambda_offset_i*cutoff);
+        du_dl_j -= FLOAT_TO_FIXED_NONBONDED(-delta_prefactor*delta_w*lambda_offset_j*cutoff);
 
         // energy is size extensive so this may not be a good idea
-        energy += FLOAT_TO_FIXED(-u);
+        energy -= FLOAT_TO_FIXED_NONBONDED(u);
 
-        g_qi += FLOAT_TO_FIXED_DU_DP<RealType, FIXED_EXPONENT_DU_DCHARGE>(-charge_scale*qj*inv_dij*ebd);
-        g_qj += FLOAT_TO_FIXED_DU_DP<RealType, FIXED_EXPONENT_DU_DCHARGE>(-charge_scale*qi*inv_dij*ebd);
+        g_qi -= FLOAT_TO_FIXED_DU_DP<RealType, FIXED_EXPONENT_DU_DCHARGE>(charge_scale*qj*inv_dij*ebd);
+        g_qj -= FLOAT_TO_FIXED_DU_DP<RealType, FIXED_EXPONENT_DU_DCHARGE>(charge_scale*qi*inv_dij*ebd);
```

Comment:
Curiosity: Do the `+= (- val)` -> `-= val` edits have an effect on speed or floating-point conversion precision?

Reply:
Compilers can optimize this away easily since ```x = x + - val``` is literally ```just x = x - val```, there may be some throughput differences negating in integer (which is done by setting all bits left of the left-most set bit to 1), vs negating a float (which sets only the left most bit to one). But this currently has to be done the way shown above for correctness.